crates/sui-types/src/messages_checkpoint.rs

```diff
@@ -771,4 +773,67 @@ mod tests {
                 .is_err()
         )
     }
+
+    // Generate a CheckpointSummary from the input transaction digest. All the other fields in the generated
+    // CheckpointSummary will be the same. The generated CheckpointSummary can be used to test how input
+    // transaction digest affects CheckpointSummary.
+    fn generate_test_checkpoint_summary_from_digest(
+        digest: TransactionDigest,
+    ) -> CheckpointSummary {
+        CheckpointSummary::new(
+            1,
+            2,
+            10,
+            &CheckpointContents::new_with_digests_only_for_tests([ExecutionDigests::new(
+                digest,
+                TransactionEffectsDigest::ZERO,
+            )]),
+            None,
+            GasCostSummary::default(),
+            None,
+            100,
+        )
+    }
+
+    // Tests that ConsensusCommitPrologue with different consensus commit digest will result in different checkpoint content.
+    #[test]
+    fn test_checkpoing_summary_with_different_consensus_digest() {
```

Comment:
This test is the best scenario that I can think of right now to test the current code without writing a complex integration test with artificial folk. It basically tests that given a different consensus output digest, the checkpoint digest will be different. Any other testing suggestions are appreciated!